image-patching/stopgap-script/nuage_overcloud_full_patch.py

```diff
@@ -216,73 +209,67 @@ def create_repo_file(reponame, repoUrl, image, gpgcheck):
 #####
 def delete_repo_file(image):
     virt_customize(
-        '"rm -f /etc/yum.repos.d/nuage.repo" -a %s --selinux-relabel'
-        % (image))
+        '"rm -f /etc/yum.repos.d/nuage.repo" -a %s --selinux-relabel' %
+        (image))
 
 
 #####
 # Function to install Nuage AVRS packages that are required
 #####
 
+
 def copy_avrs_packages(image, proxy_hostname=None, proxy_port=None):
-    if proxy_hostname != None and proxy_port != None:
-        avrs_cmds = 'cat <<EOT > nuage_avrs_packages \n' \
-                    'export http_proxy=http://%s:%s \n' \
-                    'export https_proxy=http://%s:%s \n' % (
-                    proxy_hostname, proxy_port, proxy_hostname,
-                    proxy_port)
+    if proxy_hostname is not None and proxy_port is not None:
+        avrs_cmds = '''cat <<EOT > nuage_avrs_packages
+export http_proxy=http://{}:{}
+export https_proxy=http://{}:{}'''.format(proxy_hostname, proxy_port,
+                                          proxy_hostname, proxy_port)
     else:
         avrs_cmds = 'cat <<EOT > nuage_avrs_packages \n'
 
-    install_cmds = \
-        'mkdir ./6wind \n ' \
-        'rm -rf /var/cache/yum/Nuage \n' \
-        'yum clean all \n' \
-        'touch /kernel-version \n' \
-        'rpm -q kernel | awk \'{ print substr(\$1,8) }\' ' \
-        '> /kernel-version \n' \
-        'yum install -y createrepo \n' \
-        'yum install --downloadonly --downloaddir=./6wind ' \
-        'kernel-headers-\$(cat /kernel-version) ' \
-        'kernel-devel-\$(cat /kernel-version) ' \
-        'kernel-debug-devel-\$(cat /kernel-version) ' \
-        'python-pyelftools* dkms* 6windgate* ' \
-        'nuage-openvswitch nuage-metadata-agent virtual-accelerator* \n' \
-        'yum install --downloadonly --downloaddir=./6wind ' \
-        'selinux-policy-nuage-avrs* \n' \
-        'yum install --downloadonly --downloaddir=./6wind ' \
-        '6wind-openstack-extensions \n' \
-        'rm -rf /kernel-version \n' \
-        'EOT'
+    install_cmds = '''
+#!/bin/bash
+set -xe
+mkdir -p /6wind
+rm -rf /var/cache/yum/Nuage
+yum clean all
+touch /kernel-version
+rpm -q kernel | awk '{ print substr($1,8) }' > /kernel-version
+yum install --setopt=skip_missing_names_on_install=False -y createrepo
+yum install --setopt=skip_missing_names_on_install=False --downloadonly --downloaddir=/6wind kernel-headers-$(cat /kernel-version) kernel-devel-$(cat /kernel-version) kernel-debug-devel-$(cat /kernel-version) python-pyelftools* dkms* 6windgate* nuage-openvswitch nuage-metadata-agent virtual-accelerator* 
```

Comment:
Actual output is different from expected output

Actual output:
```
yum install --setopt=skip_missing_names_on_install=False --downloadonly --downloaddir=/6wind kernel-headers-3.10.0-229.el7.x86_64
3.10.0-862.11.6.el7.x86_64
3.10.0-957.el7.x86_64 kernel-devel-3.10.0-229.el7.x86_64
3.10.0-862.11.6.el7.x86_64
3.10.0-957.el7.x86_64 kernel-debug-devel-3.10.0-229.el7.x86_64
3.10.0-862.11.6.el7.x86_64
3.10.0-957.el7.x86_64 python-pyelftools* dkms* 6windgate* nuage-openvswitch nuage-metadata-agent virtual-accelerator*
```

Expected Output:
```
yum install --setopt=skip_missing_names_on_install=False --downloadonly --downloaddir=/6wind kernel-headers-3.10.0-957.10.1.el7.x86_64 kernel-devel-3.10.0-957.10.1.el7.x86_64 kernel-debug-devel-3.10.0-957.10.1.el7.x86_64 python-pyelftools* dkms* 6windgate* nuage-openvswitch nuage-metadata-agent virtual-accelerator*
```

Reply:
After debug I came to know, $(cat /kernel-version) is getting the value from hypervisor but not overcloud image.

Ex:
On hypervisor
```
[root@rhdirector avrs]# cat /kernel-version
3.10.0-229.el7.x86_64
3.10.0-862.11.6.el7.x86_64
3.10.0-957.el7.x86_64
```

On Overcloud-full.qcow2
```
><fs> cat /kernel-version

```

Reply:
I created avrs repo without nuage-metedata-agent and ran the patching. 
Script didn't failed at this point and now the image doesn't have nuage-metadata-agent which is not expected behaviour